project/TestParallelization.scala

```diff
@@ -107,25 +116,38 @@ object TestParallelization {
     def testGroups: List[Tests.Group]
   }
 
-  class SimpleHashStrategy private(groups: Map[Int, Tests.Group]) extends GroupingStrategy {
+  class SimpleHashStrategy private(
+      groups: Map[Int, Tests.Group],
+      shard: Option[Int]) extends GroupingStrategy {
 
     lazy val testGroups = groups.values.toList
     val groupCount = groups.size
 
     override def add(testDefinition: TestDefinition): GroupingStrategy = {
       val groupIdx = math.abs(testDefinition.name.hashCode % groupCount)
+      if (shard.isDefined) {
```

Comment:
if we wanted to disable this, say 6 months from now due to some crazy bug, the minimal change we would have to do is just disable this if statement, right? e.g. `if (false)`, right?

Reply:
Correct, that's def a way to do it in the code itself. From the GH configuration perspective, it's possible to also just comment out the 2 lines for matrix shard ID list and the shard ID env variable.

Reply:
@amogh-jahagirdar - please include this "how to disable later" in the PR description. That will make it super findable.